connectors/src/monitoring/index.ts

```diff
@@ -0,0 +1,45 @@
+export class MonitoringError extends Error {}
+import StatsDClient from "hot-shots";
+
+import mainlogger from "@connectors/logger/logger";
+const logger = mainlogger.child({ module: "temporal-monitoring" });
+
+export const statsDClient = new StatsDClient({});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const withMonitoring = <T extends Array<any>, U>(
+  fn: (...args: T) => U
+) => {
+  return async (...args: T): Promise<U> => {
+    let result: U | undefined = undefined;
+
+    try {
+      const startTime = new Date();
+      logger.info({ activity: fn.name }, `Activity started`);
+
+      const tags = [`activity:${fn.name}`];
+      statsDClient.increment("activities.count", 1, tags);
+
+      result = await fn(...args);
+      const endTime = new Date();
+      const elapsedMs = endTime.getTime() - startTime.getTime();
+      statsDClient.histogram(`activity.duration`, elapsedMs, tags);
+      logger.info(
+        { activity: fn.name, elapsedMs: elapsedMs },
+        `Activity ended successfully`
+      );
+      return result;
+    } catch (e) {
+      let tags = [`activity:${fn.name}`];
+      if (e instanceof MonitoringError) {
+        tags = tags.concat([`known_error:true`]);
+      } else {
+        tags = tags.concat([`activity:${fn.name}`, `known_error:false`]);
+      }
+      statsDClient.increment("activities_errors.count", 1, tags);
+      logger.error({ activity: fn.name }, `activity errored`);
```

Comment:
I think we want to be as consistent in message format. Let's try to have sentences so they start with a capitalized letter and finish with a `.`. I won't be picky about the `.` but since 10 lines above you do use capitalization I'd expect to see it here as well.